micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -613,15 +617,21 @@ private Meter getOrCreateMeter(@Nullable DistributionStatisticConfig config,
             Id mappedId, Function<Meter.Id, ? extends Meter> noopBuilder) {
         Meter m = meterMap.get(mappedId);
 
-        if (m == null) {
+        if (m != null) {
+            checkAndWarnAboutGaugeDoubleRegistration(m);
+        }
+        else {
             if (isClosed()) {
                 return noopBuilder.apply(mappedId);
             }
 
             synchronized (meterMapLock) {
                 m = meterMap.get(mappedId);
 
-                if (m == null) {
+                if (m != null) {
+                    checkAndWarnAboutGaugeDoubleRegistration(m);
```

Comment:
I don't know either: JIT can optimize/de-optimize `null` checks, not sure what it can do with `instanceof` but I think it's simpler if we remove the extra null check.